vic/drivers/cesm/src/cesm_interface_c.c

```diff
@@ -0,0 +1,181 @@
+nc_file_struct      nc_hist_file;
+nc_var_struct       nc_vars[N_OUTVAR_TYPES];
+option_struct       options;
+parameters_struct   param;
+out_data_struct   **out_data;
+save_data_struct   *save_data;
+param_set_struct    param_set;
+soil_con_struct    *soil_con = NULL;
+veg_con_map_struct *veg_con_map = NULL;
+veg_con_struct    **veg_con = NULL;
+veg_hist_struct   **veg_hist = NULL;
+veg_lib_struct    **veg_lib = NULL;
+
+/******************************************************************************
+ * @brief    Initialize VIC MPI communicator
+ *****************************************************************************/
+int
+vic_cesm_init_mpi(int MPI_COMM_VIC_F)
+{
+    // Set the VIC MPI communicator
+    MPI_COMM_VIC = MPI_Comm_f2c(MPI_COMM_VIC_F);
+
+    return 0;
+}
+
+/******************************************************************************
+ * @brief    Initialization function for CESM driver
+ *****************************************************************************/
+int
+vic_cesm_init(char     *vic_global_param_file,
+              char     *caseid,
+              char     *runtype,
+              vic_clock vclock)
+{
+    // Initialize Log Destination
+    initialize_log();
+
+    // read global parameters
+    vic_start();
+
+    if (mpi_rank == 0) {
+        // Set driver specific filenames and paths
+        strcpy(filenames.global, vic_global_param_file);
+        strcpy(filenames.log_path, "");  // write log files to cwd
+        strcpy(filenames.result_dir, ""); // write result files to cwd
+
+        // caseid
+        strcpy(global_param.caseid, caseid);
+
+        // Unpack the vic_clock structure
+        // Model timestep
+        global_param.dt = (double) vclock.timestep;
+        global_param.model_steps_per_day =
+            (int) ((double) SEC_PER_DAY / global_param.dt);
+
+        // Start date/time
+        global_param.startyear = vclock.current_year;
+        global_param.startmonth = vclock.current_month;
+        global_param.startday = vclock.current_day;
+        global_param.startsec = vclock.current_dayseconds;
+        global_param.nrecs = 1;
+
+        // Calendar
+        global_param.calendar = calendar_from_char(vclock.calendar);
+    }
+
+    // Initialize time
+    initialize_cesm_time();
+
+    // allocate memory
+    vic_alloc();
+
+    // initialize model parameters from parameter files
+    vic_init();
+
+    // restore model state, either using a cold start or from a restart file
+    vic_restore(runtype);
+
+    // initialize output structures
+    vic_init_output();
+
+    return 0;
+}
+
+/******************************************************************************
+ * @brief    Run function for CESM driver
+ *****************************************************************************/
+int
+vic_cesm_run(vic_clock vclock)
+{
+    // reset l2x fields
+    initialize_l2x_data();
+
+    // advance the clock
+    advance_time();
+    assert_time_insync(&vclock, &dmy);
```

Comment:
It's fine on the c side.  As long as it's clear that it's happening and is working correctly.